### PR TITLE
Add TI dra7xx platform support

### PR DESCRIPTION
**Pull request by HarinarayanaBhatta**
Added plat-ti, with initial support for TI dra7xx platform.

Changed generic init_sec_mon to be overriden by platform/main.c
because initial return to non-secure world needs to restore full
NS context for this platform.

Signed-off-by: Harinarayan Bhatta <harinarayan.bhatta@linaro.org>

### CODE REVIEW

File: core/arch/arm/plat-ti/platform_config.h [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+#define CFG_SHMEM_SIZE		0x100000
+
+#define CFG_TEE_RAM_VA_SIZE	(1024 * 1024)
+
+#ifndef CFG_TEE_LOAD_ADDR
+#define CFG_TEE_LOAD_ADDR	(CFG_TEE_RAM_START + 0x100)
+#endif
+
+#ifdef CFG_WITH_PAGER
+/*
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-ti/platform_config.h]: Do you plan to ever enable paging for this platform? If not you can remove this.

File: core/arch/arm/plat-ti/platform_config.h [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ * Copyright (c) 2014, Linaro Limited
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-ti/platform_config.h]: It's 2015 now. :-)

File: core/arch/arm/plat-ti/platform_config.h [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	(PLATFORM_FLAVOR == PLATFORM_FLAVOR_ID_ ## flav)
+
+#if PLATFORM_FLAVOR_IS(dra7xx)
+
+#define DRAM0_BASE		0x94C00000
+#define DRAM0_SIZE		0x00800000
+
+#ifdef CFG_WITH_PAGER
+#error Pager not supported on this platform
+#else /*CFG_WITH_PAGER*/
```
Comment (jforissier) [inline core/arch/arm/plat-ti/platform_config.h]: `#endif` here and remove `#ifndef CFG_WITH_PAGER` below

File: core/arch/arm/plat-ti/platform_config.h [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+/* Location of trusted dram on the base fvp */
```
Comment (jforissier) [inline core/arch/arm/plat-ti/platform_config.h]: fvp?

File: core/arch/arm/plat-ti/platform_flags.mk [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+DEBUG		?= 1
+ifeq ($(DEBUG),1)
+platform-cflags += -O0
+else
+platform-cflags += -Os
+endif
+
+platform-cflags += -g
+platform-aflags += -g
```
Comment (jforissier) [inline core/arch/arm/plat-ti/platform_flags.mk]: Remove these lines since you have `-g3` below.

File: core/arch/arm/plat-ti/conf.mk [hunk 6]
```diff
@@ -0,0 +1,27 @@
+include core/arch/$(ARCH)/plat-$(PLATFORM)/platform_flags.mk
+
+CROSS_COMPILE	?= arm-linux-gnueabihf-
+COMPILER	?= gcc
```
Comment (jforissier) [inline core/arch/arm/plat-ti/conf.mk]: Both lines should not be needed because these are the default values set in `mk/config.mk`

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jbech-linaro** [comment]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by pascal-brand-st-dev** [comment]
@HarinarayanaBhatta 
Hello, Feel free to squash and rebase on top of master and add the tags. I'll then merge.
Thanks, Pascal.

**Comment by pascal-brand-st** [comment]
Merged manually

**Pull request closed without merging**